presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotPushdownUtils.java

```diff
@@ -147,10 +154,25 @@ public ExpressionType getExpressionType()
             if (agg != null) {
                 if (agg.getFilter().isPresent()
                         || agg.isDistinct()
-                        || agg.getOrderBy().isPresent()
-                        || agg.getMask().isPresent()) {
+                        || agg.getOrderBy().isPresent()) {
+                    throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), "Unsupported aggregation node " + aggregationNode);
+                }
+                if (agg.getMask().isPresent()) {
+                    /**
+                     * This if-block handles the case of pushing down distinct count as one of the aggregation expression along with other aggregation functions.
+                     * E.g. `SELECT count(distinct COL_A), sum(COL_B) FROM (SELECT COL_A, COL_B FROM myTable)` to Pinot as `SELECT distinctCount(COL_A), sum(COL_B) FROM myTable`
+                     *
+                     */
+                    if (agg.getCall().getDisplayName().equalsIgnoreCase(COUNT_FUNCTION_NAME) && agg.getMask().get().getName().equalsIgnoreCase(agg.getArguments().get(0) + DISTINCT_MASK)) {
+                        nodeBuilder.add(new AggregationFunctionColumnNode(outputColumn, new CallExpression(DISTINCT_COUNT_FUNCTION_NAME, agg.getCall().getFunctionHandle(), agg.getCall().getType(), agg.getCall().getArguments())));
+                        continue;
+                    }
+                    // Pinot doesn't support push down aggregation functions other than count on top of distinct function.
                     throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), "Unsupported aggregation node " + aggregationNode);
```

Comment:
Will this message show the agg mask too or reflect that we have a distinct operation ?

Reply:
Both. We only handle distinct mask + count function, which means it will throw exception on  `sum(distinct a)` or `count` with other masks